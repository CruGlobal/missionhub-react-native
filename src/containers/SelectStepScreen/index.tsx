import React from 'react';
import { SafeAreaView, View } from 'react-native';
import { useSelector, useDispatch } from 'react-redux';
import { ThunkDispatch, ThunkAction } from 'redux-thunk';
// eslint-disable-next-line import/default
import ParallaxScrollView from 'react-native-parallax-scroll-view';

import { Text } from '../../components/common';
import BackButton from '../BackButton';
import Skip from '../../components/Skip';
import theme from '../../theme';
import StepsList from '../StepsList';
import Header from '../../components/Header';
import { useAnalytics } from '../../utils/hooks/useAnalytics';
import { personSelector } from '../../selectors/people';
import { PeopleState, Person } from '../../reducers/people';

import styles from './styles';

export interface Step {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  [key: string]: any;
}

interface SelectStepScreenProps {
  personId: string;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  orgId?: string;
  contactStageId: string;
  headerText: [string, string];
  enableSkipButton?: boolean;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  next: (nextProps: {
    personId: string;
    step?: Step;
    skip: boolean;
    orgId?: string;
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
  }) => ThunkAction<void, any, null, never>;
}

const SelectStepScreen = ({
  personId,
  orgId,
  contactStageId,
  headerText,
  enableSkipButton = false,
  next,
}: SelectStepScreenProps) => {
  useAnalytics('add step');
  const dispatch = useDispatch();

  const person = useSelector<{ people: PeopleState }, Person>(({ people }) =>
    personSelector({ people }, { personId, orgId }),
  );

  const navigateNext = (step?: Step, skip = false) => {
    dispatch(
      next({
        personId,
        step,
        skip,
        orgId,
      }),
    );
  };

  const navToSuggestedStep = (step: Step) => {
    navigateNext(step);
  };

  const navToCreateStep = () => {
    navigateNext();
  };

  const handleSkip = () => {
    navigateNext(undefined, true);
  };

  const renderForeground = () => {
    return (
      <View style={{ flex: 1 }}>
        {renderHeader()}
        <View style={{ flex: 1, alignItems: 'center' }}>
          <Text style={styles.headerText}>{headerText[0]}</Text>
          <Text style={styles.headerText}>{headerText[1]}</Text>
        </View>
      </View>
    );
  };

  const renderHeader = () => (
    <Header
      left={<BackButton />}
      right={enableSkipButton && <Skip onSkip={handleSkip} />}
    />
  );

  const { headerHeight } = theme;

  return (
    <View style={styles.container}>
      <SafeAreaView style={{ backgroundColor: theme.primaryColor }} />
      <ParallaxScrollView
        backgroundColor={theme.primaryColor}
        contentBackgroundColor={theme.extraLightGrey}
        parallaxHeaderHeight={180}
        renderForeground={renderForeground}
        stickyHeaderHeight={headerHeight}
        renderStickyHeader={renderHeader}
      >
        <StepsList
          onPressCreateStep={navToCreateStep}
          contactName={person && person.first_name}
          personId={personId}
          contactStageId={contactStageId}
          onPressStep={navToSuggestedStep}
        />
      </ParallaxScrollView>
    </View>
  );
};

export default SelectStepScreen;
