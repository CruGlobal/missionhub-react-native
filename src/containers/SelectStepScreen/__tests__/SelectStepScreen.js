import 'react-native';
import React from 'react';

import SelectStepScreen, { mapStateToProps } from '..';

import {
  renderShallow,
  createMockStore,
  testSnapshotShallow,
} from '../../../../testUtils';
import * as navigation from '../../../actions/navigation';
import { ADD_STEP_SCREEN } from '../../AddStepScreen';
import { getStepSuggestions } from '../../../actions/steps';
import { shuffleArray } from '../../../utils/common';
import { CREATE_STEP } from '../../../constants';

jest.mock('react-native-device-info');
jest.mock('../../../actions/steps');
jest.mock('../../../utils/common');
shuffleArray.mockImplementation(arr => arr);

const testName = 'Bill';

const contactStageId = 5;

const suggestions = [
  { id: '1', body: 'test 1' },
  { id: '2', body: 'test 2' },
  { id: '3', body: 'test 3' },
  { id: '4', body: 'test 4' },
  { id: '5', body: 'test 5' },
  { id: '6', body: 'test 6' },
  { id: '7', body: 'test 7' },
  { id: '8', body: 'test 8' },
  { id: '9', body: 'test 9' },
];

const auth = {
  person: {
    id: '123',
  },
};

const org = { id: 2 };

const receiverId = '1';

const headerText = 'header';

const steps = {
  suggestedForOthers: {
    [contactStageId]: suggestions,
  },
};

let store = createMockStore({
  auth,
  steps,
});

let component, parallaxProps, flatListProps, instance;
let onComplete = jest.fn();
let createStepTracking = {};
let enableBackButton = false;

const createComponent = async () => {
  getStepSuggestions.mockReturnValue(
    Promise.resolve({ response: suggestions }),
  );
  component = renderShallow(
    <SelectStepScreen
      isMe={false}
      headerText={headerText}
      contactStageId={contactStageId}
      createStepTracking={createStepTracking}
      onComplete={onComplete}
      personFirstName={testName}
      enableBackButton={enableBackButton}
      receiverId={receiverId}
      organization={org}
    />,
    store,
  );
  parallaxProps = component.find('ParallaxScrollView').props();
  flatListProps = component.find('FlatList').props();
  instance = component.instance();
  await Promise.resolve();
};

describe('mapStateToProps', () => {
  it('should provide necessary props for me', () => {
    expect(
      mapStateToProps(
        {
          auth: {
            person: {
              id: '123',
            },
          },
          steps: {
            suggestedForMe: {
              [contactStageId]: [{ id: '1', body: 'test 1' }],
            },
            suggestedForOthers: {},
          },
        },
        {
          isMe: true,
          contactStageId,
        },
      ),
    ).toMatchSnapshot();
  });

  it('should provide necessary props for others', () => {
    expect(
      mapStateToProps(
        {
          auth: {
            person: {
              id: '123',
            },
          },
          steps: {
            suggestedForMe: {},
            suggestedForOthers: {
              [contactStageId]: [{ id: '1', body: 'test 1' }],
            },
          },
        },
        {
          isMe: false,
          contactStageId,
        },
      ),
    ).toMatchSnapshot();
  });
});

describe('SelectStepScreen', () => {
  beforeEach(async () => {
    await createComponent();
  });

  it('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render foreground header correctly', () => {
    testSnapshotShallow(parallaxProps.renderForeground());
  });

  it('should render sticky header correctly', () => {
    testSnapshotShallow(parallaxProps.renderStickyHeader());
  });

  it('should render step item correctly', () => {
    testSnapshotShallow(flatListProps.renderItem({ item: suggestions[0] }));
  });

  it('should render load more button correctly', () => {
    testSnapshotShallow(flatListProps.ListFooterComponent());
  });

  it('should call navigate back two screens', () => {
    navigation.navigateBack = jest.fn();
    instance.navigateBackTwoScreens();
    expect(navigation.navigateBack).toHaveBeenCalledWith(2);
  });

  it('should call steps list ref', () => {
    instance.stepsListRef('test');
    expect(instance.stepsList).toEqual('test');
  });
});

describe('componentDidMount', () => {
  it('should call getStepSuggestions every time component mounts', async () => {
    store = createMockStore({ auth, steps: { suggestedForOthers: {} } });
    await createComponent();

    expect(getStepSuggestions).toHaveBeenCalled();
    store = createMockStore({ auth, steps });
  });
});

describe('renderBackButton', () => {
  it('should render back button', async () => {
    enableBackButton = true;
    await createComponent();

    expect(component).toMatchSnapshot();
    enableBackButton = false;
  });
});

describe('Navigation', () => {
  it('navigates to add step screen', async () => {
    navigation.navigatePush = jest.fn();
    createStepTracking = { test: 'this is a test tracking property' };
    onComplete = jest.fn();
    await createComponent();

    instance.handleCreateStep();

    expect(navigation.navigatePush).toHaveBeenCalledWith(ADD_STEP_SCREEN, {
      type: CREATE_STEP,
      onComplete: expect.any(Function),
      trackingObj: createStepTracking,
    });
    createStepTracking = {};
    onComplete = jest.fn();
  });
});

describe('handleLoadSteps', () => {
  beforeAll(async () => {
    await createComponent();
  });

  it('Initially displays four suggestions', () => {
    expect(instance.state.steps).toEqual(suggestions.slice(0, 4));
  });
  it('Loads four more suggestions', () => {
    instance.handleLoadSteps();
    component.update();
    expect(instance.state.steps).toEqual(suggestions.slice(0, 8));
  });
  it('Loads last suggestion', () => {
    instance.handleLoadSteps();
    component.update();
    expect(instance.state.steps).toEqual(suggestions);
  });
  it('loads no more because all suggestions are displayed', () => {
    instance.handleLoadSteps();
    component.update();
    expect(instance.state.steps).toEqual(suggestions);
  });
});
