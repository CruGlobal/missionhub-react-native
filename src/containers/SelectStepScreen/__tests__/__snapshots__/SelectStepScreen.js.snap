// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with enableBackButton renders correctly 1`] = `
<View
  flex={1}
>
  <View
    onLayout={[Function]}
    style={
      Array [
        undefined,
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#007398",
          "height": 215,
          "opacity": 1,
          "overflow": "hidden",
          "position": "absolute",
          "top": 0,
          "transform": Array [
            Object {
              "translateY": 0,
            },
            Object {
              "scale": 1,
            },
          ],
          "width": 750,
        }
      }
    >
      <View>
        <View />
      </View>
    </View>
    <RCTScrollView
      onChangeHeaderVisibility={[Function]}
      onScroll={[Function]}
      scrollEventThrottle={1}
      style={
        Object {
          "backgroundColor": "transparent",
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "backgroundColor": "transparent",
              "overflow": "hidden",
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "transparent",
                "height": 215,
                "opacity": 1,
                "overflow": "hidden",
              }
            }
          >
            <View
              style={
                Object {
                  "height": 215,
                }
              }
            >
              <View
                alignItems="center"
                flex={1}
              >
                <RCTSafeAreaView
                  emulateUnlessSupported={true}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#007398",
                        "flexDirection": "row",
                        "height": 65,
                        "paddingTop": 20,
                      },
                      undefined,
                      null,
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "paddingLeft": 5,
                        },
                        Object {
                          "alignItems": "flex-start",
                          "flex": 1,
                          "justifyContent": "center",
                        },
                      ]
                    }
                  />
                  <View
                    style={
                      Array [
                        Object {
                          "paddingRight": 5,
                        },
                        Object {
                          "alignItems": "flex-end",
                          "flex": 1,
                          "justifyContent": "center",
                        },
                      ]
                    }
                  />
                </RCTSafeAreaView>
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 12,
                      },
                      Array [
                        Object {
                          "color": "#ffffff",
                        },
                        Object {
                          "fontSize": 18,
                        },
                        Object {
                          "color": "#52C5DC",
                          "fontSize": 40,
                        },
                      ],
                      Object {
                        "fontFamily": "icomoon",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  î¤‚
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#505256",
                      },
                      Object {
                        "fontFamily": "AmaticSC-Bold",
                      },
                      Object {
                        "color": "#52C5DC",
                        "fontSize": 36,
                        "letterSpacing": 2,
                        "lineHeight": 48,
                      },
                    ]
                  }
                >
                  add steps of faith
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#505256",
                      },
                      Object {
                        "fontFamily": "SourceSansPro-Regular",
                      },
                      Object {
                        "color": "#ffffff",
                        "fontSize": 16,
                        "lineHeight": 24,
                        "paddingBottom": 36,
                        "paddingHorizontal": 50,
                        "textAlign": "center",
                      },
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Array [
                Object {
                  "backgroundColor": "#ECEEF2",
                },
              ],
              Object {
                "minHeight": 0,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "display": "none",
              }
            }
          />
          <StepsList
            contactName="roger"
            contactStageId="3"
            onPressStep={[Function]}
            organization={
              Object {
                "id": "4234234",
              }
            }
            receiverId="252342354234"
          />
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#ECEEF2",
            }
          }
        />
      </View>
    </RCTScrollView>
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "height": 65,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "top": 0,
          "transform": Array [
            Object {
              "translateY": -65,
            },
          ],
          "width": 750,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#007398",
            "height": 65,
            "opacity": 0,
            "transform": Array [
              Object {
                "translateY": -65,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "transform": Array [
                Object {
                  "translateY": 65,
                },
              ],
            }
          }
        >
          <RCTSafeAreaView
            emulateUnlessSupported={true}
            style={
              Array [
                Object {
                  "backgroundColor": "#007398",
                  "flexDirection": "row",
                  "height": 65,
                  "paddingTop": 20,
                },
                undefined,
                Object {
                  "elevation": 4,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "paddingLeft": 5,
                  },
                  Object {
                    "alignItems": "flex-start",
                    "flex": 1,
                    "justifyContent": "center",
                  },
                ]
              }
            />
            <View
              style={
                Array [
                  Object {},
                  Object {
                    "alignItems": "center",
                    "flex": 5,
                    "justifyContent": "center",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#505256",
                    },
                    Object {
                      "fontFamily": "SourceSansPro-Regular",
                    },
                    Object {
                      "color": "#ffffff",
                      "fontSize": 14,
                    },
                  ]
                }
              >
                ADD STEPS OF FAITH
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "paddingRight": 5,
                  },
                  Object {
                    "alignItems": "flex-end",
                    "flex": 1,
                    "justifyContent": "center",
                  },
                ]
              }
            />
          </RCTSafeAreaView>
        </View>
      </View>
    </View>
  </View>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="BottomButton"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#52C5DC",
              "height": 60,
              "justifyContent": "center",
            },
            null,
            Object {
              "width": 750,
            },
            null,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#505256",
              },
              Object {
                "fontFamily": "SourceSansPro-Regular",
              },
              Array [
                Object {
                  "backgroundColor": "rgba(0,0,0,0)",
                  "color": "#ffffff",
                  "fontSize": 18,
                  "fontWeight": "500",
                  "letterSpacing": 1.5,
                  "textAlign": "center",
                },
                Object {},
              ],
            ]
          }
        >
          CREATE YOUR OWN STEP
        </Text>
      </View>
    </View>
  </RCTSafeAreaView>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "alignItems": "flex-start",
          "alignSelf": "flex-start",
          "justifyContent": "center",
        },
        undefined,
        Object {
          "left": 5,
          "position": "absolute",
          "top": 25,
        },
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="IconButton"
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            null,
            Object {},
            null,
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {
                  "color": "#ffffff",
                },
                Object {
                  "fontSize": 18,
                },
                Array [
                  Object {
                    "margin": 10,
                  },
                  Object {},
                ],
              ],
              Object {
                "fontFamily": "icomoon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          î¤ƒ
        </Text>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;

exports[`with enableSkipButton renders correctly 1`] = `
<View
  flex={1}
>
  <View
    onLayout={[Function]}
    style={
      Array [
        undefined,
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#007398",
          "height": 215,
          "opacity": 1,
          "overflow": "hidden",
          "position": "absolute",
          "top": 0,
          "transform": Array [
            Object {
              "translateY": 0,
            },
            Object {
              "scale": 1,
            },
          ],
          "width": 750,
        }
      }
    >
      <View>
        <View />
      </View>
    </View>
    <RCTScrollView
      onChangeHeaderVisibility={[Function]}
      onScroll={[Function]}
      scrollEventThrottle={1}
      style={
        Object {
          "backgroundColor": "transparent",
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "backgroundColor": "transparent",
              "overflow": "hidden",
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "transparent",
                "height": 215,
                "opacity": 1,
                "overflow": "hidden",
              }
            }
          >
            <View
              style={
                Object {
                  "height": 215,
                }
              }
            >
              <View
                alignItems="center"
                flex={1}
              >
                <RCTSafeAreaView
                  emulateUnlessSupported={true}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#007398",
                        "flexDirection": "row",
                        "height": 65,
                        "paddingTop": 20,
                      },
                      undefined,
                      null,
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "paddingLeft": 5,
                        },
                        Object {
                          "alignItems": "flex-start",
                          "flex": 1,
                          "justifyContent": "center",
                        },
                      ]
                    }
                  />
                  <View
                    style={
                      Array [
                        Object {
                          "paddingRight": 5,
                        },
                        Object {
                          "alignItems": "flex-end",
                          "flex": 1,
                          "justifyContent": "center",
                        },
                      ]
                    }
                  />
                </RCTSafeAreaView>
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 12,
                      },
                      Array [
                        Object {
                          "color": "#ffffff",
                        },
                        Object {
                          "fontSize": 18,
                        },
                        Object {
                          "color": "#52C5DC",
                          "fontSize": 40,
                        },
                      ],
                      Object {
                        "fontFamily": "icomoon",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  î¤‚
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#505256",
                      },
                      Object {
                        "fontFamily": "AmaticSC-Bold",
                      },
                      Object {
                        "color": "#52C5DC",
                        "fontSize": 36,
                        "letterSpacing": 2,
                        "lineHeight": 48,
                      },
                    ]
                  }
                >
                  add steps of faith
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#505256",
                      },
                      Object {
                        "fontFamily": "SourceSansPro-Regular",
                      },
                      Object {
                        "color": "#ffffff",
                        "fontSize": 16,
                        "lineHeight": 24,
                        "paddingBottom": 36,
                        "paddingHorizontal": 50,
                        "textAlign": "center",
                      },
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Array [
                Object {
                  "backgroundColor": "#ECEEF2",
                },
              ],
              Object {
                "minHeight": 0,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "display": "none",
              }
            }
          />
          <StepsList
            contactName="roger"
            contactStageId="3"
            onPressStep={[Function]}
            organization={
              Object {
                "id": "4234234",
              }
            }
            receiverId="252342354234"
          />
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#ECEEF2",
            }
          }
        />
      </View>
    </RCTScrollView>
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "height": 65,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "top": 0,
          "transform": Array [
            Object {
              "translateY": -65,
            },
          ],
          "width": 750,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#007398",
            "height": 65,
            "opacity": 0,
            "transform": Array [
              Object {
                "translateY": -65,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "transform": Array [
                Object {
                  "translateY": 65,
                },
              ],
            }
          }
        >
          <RCTSafeAreaView
            emulateUnlessSupported={true}
            style={
              Array [
                Object {
                  "backgroundColor": "#007398",
                  "flexDirection": "row",
                  "height": 65,
                  "paddingTop": 20,
                },
                undefined,
                Object {
                  "elevation": 4,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "paddingLeft": 5,
                  },
                  Object {
                    "alignItems": "flex-start",
                    "flex": 1,
                    "justifyContent": "center",
                  },
                ]
              }
            />
            <View
              style={
                Array [
                  Object {},
                  Object {
                    "alignItems": "center",
                    "flex": 5,
                    "justifyContent": "center",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#505256",
                    },
                    Object {
                      "fontFamily": "SourceSansPro-Regular",
                    },
                    Object {
                      "color": "#ffffff",
                      "fontSize": 14,
                    },
                  ]
                }
              >
                ADD STEPS OF FAITH
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "paddingRight": 5,
                  },
                  Object {
                    "alignItems": "flex-end",
                    "flex": 1,
                    "justifyContent": "center",
                  },
                ]
              }
            />
          </RCTSafeAreaView>
        </View>
      </View>
    </View>
  </View>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="BottomButton"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#52C5DC",
              "height": 60,
              "justifyContent": "center",
            },
            null,
            Object {
              "width": 750,
            },
            null,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#505256",
              },
              Object {
                "fontFamily": "SourceSansPro-Regular",
              },
              Array [
                Object {
                  "backgroundColor": "rgba(0,0,0,0)",
                  "color": "#ffffff",
                  "fontSize": 18,
                  "fontWeight": "500",
                  "letterSpacing": 1.5,
                  "textAlign": "center",
                },
                Object {},
              ],
            ]
          }
        >
          CREATE YOUR OWN STEP
        </Text>
      </View>
    </View>
  </RCTSafeAreaView>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "position": "absolute",
        "right": 10,
        "top": 0,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="SkipButton"
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            null,
            Array [
              Object {
                "padding": 15,
              },
              undefined,
            ],
            null,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#505256",
              },
              Object {
                "fontFamily": "SourceSansPro-Regular",
              },
              Array [
                Object {
                  "backgroundColor": "rgba(0,0,0,0)",
                  "color": "#ffffff",
                  "fontSize": 18,
                  "fontWeight": "500",
                  "letterSpacing": 1.5,
                  "textAlign": "center",
                },
                Array [
                  Object {
                    "color": "#ffffff",
                  },
                  undefined,
                ],
              ],
            ]
          }
        >
          SKIP
        </Text>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;

exports[`without enableBackButton nor enableSkipButton renders correctly 1`] = `
<View
  flex={1}
>
  <View
    onLayout={[Function]}
    style={
      Array [
        undefined,
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#007398",
          "height": 215,
          "opacity": 1,
          "overflow": "hidden",
          "position": "absolute",
          "top": 0,
          "transform": Array [
            Object {
              "translateY": 0,
            },
            Object {
              "scale": 1,
            },
          ],
          "width": 750,
        }
      }
    >
      <View>
        <View />
      </View>
    </View>
    <RCTScrollView
      onChangeHeaderVisibility={[Function]}
      onScroll={[Function]}
      scrollEventThrottle={1}
      style={
        Object {
          "backgroundColor": "transparent",
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "backgroundColor": "transparent",
              "overflow": "hidden",
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "transparent",
                "height": 215,
                "opacity": 1,
                "overflow": "hidden",
              }
            }
          >
            <View
              style={
                Object {
                  "height": 215,
                }
              }
            >
              <View
                alignItems="center"
                flex={1}
              >
                <RCTSafeAreaView
                  emulateUnlessSupported={true}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#007398",
                        "flexDirection": "row",
                        "height": 65,
                        "paddingTop": 20,
                      },
                      undefined,
                      null,
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "paddingLeft": 5,
                        },
                        Object {
                          "alignItems": "flex-start",
                          "flex": 1,
                          "justifyContent": "center",
                        },
                      ]
                    }
                  />
                  <View
                    style={
                      Array [
                        Object {
                          "paddingRight": 5,
                        },
                        Object {
                          "alignItems": "flex-end",
                          "flex": 1,
                          "justifyContent": "center",
                        },
                      ]
                    }
                  />
                </RCTSafeAreaView>
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 12,
                      },
                      Array [
                        Object {
                          "color": "#ffffff",
                        },
                        Object {
                          "fontSize": 18,
                        },
                        Object {
                          "color": "#52C5DC",
                          "fontSize": 40,
                        },
                      ],
                      Object {
                        "fontFamily": "icomoon",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  î¤‚
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#505256",
                      },
                      Object {
                        "fontFamily": "AmaticSC-Bold",
                      },
                      Object {
                        "color": "#52C5DC",
                        "fontSize": 36,
                        "letterSpacing": 2,
                        "lineHeight": 48,
                      },
                    ]
                  }
                >
                  add steps of faith
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#505256",
                      },
                      Object {
                        "fontFamily": "SourceSansPro-Regular",
                      },
                      Object {
                        "color": "#ffffff",
                        "fontSize": 16,
                        "lineHeight": 24,
                        "paddingBottom": 36,
                        "paddingHorizontal": 50,
                        "textAlign": "center",
                      },
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Array [
                Object {
                  "backgroundColor": "#ECEEF2",
                },
              ],
              Object {
                "minHeight": 0,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "display": "none",
              }
            }
          />
          <StepsList
            contactName="roger"
            contactStageId="3"
            onPressStep={[Function]}
            organization={
              Object {
                "id": "4234234",
              }
            }
            receiverId="252342354234"
          />
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#ECEEF2",
            }
          }
        />
      </View>
    </RCTScrollView>
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "height": 65,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "top": 0,
          "transform": Array [
            Object {
              "translateY": -65,
            },
          ],
          "width": 750,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#007398",
            "height": 65,
            "opacity": 0,
            "transform": Array [
              Object {
                "translateY": -65,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "transform": Array [
                Object {
                  "translateY": 65,
                },
              ],
            }
          }
        >
          <RCTSafeAreaView
            emulateUnlessSupported={true}
            style={
              Array [
                Object {
                  "backgroundColor": "#007398",
                  "flexDirection": "row",
                  "height": 65,
                  "paddingTop": 20,
                },
                undefined,
                Object {
                  "elevation": 4,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "paddingLeft": 5,
                  },
                  Object {
                    "alignItems": "flex-start",
                    "flex": 1,
                    "justifyContent": "center",
                  },
                ]
              }
            />
            <View
              style={
                Array [
                  Object {},
                  Object {
                    "alignItems": "center",
                    "flex": 5,
                    "justifyContent": "center",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#505256",
                    },
                    Object {
                      "fontFamily": "SourceSansPro-Regular",
                    },
                    Object {
                      "color": "#ffffff",
                      "fontSize": 14,
                    },
                  ]
                }
              >
                ADD STEPS OF FAITH
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "paddingRight": 5,
                  },
                  Object {
                    "alignItems": "flex-end",
                    "flex": 1,
                    "justifyContent": "center",
                  },
                ]
              }
            />
          </RCTSafeAreaView>
        </View>
      </View>
    </View>
  </View>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="BottomButton"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#52C5DC",
              "height": 60,
              "justifyContent": "center",
            },
            null,
            Object {
              "width": 750,
            },
            null,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#505256",
              },
              Object {
                "fontFamily": "SourceSansPro-Regular",
              },
              Array [
                Object {
                  "backgroundColor": "rgba(0,0,0,0)",
                  "color": "#ffffff",
                  "fontSize": 18,
                  "fontWeight": "500",
                  "letterSpacing": 1.5,
                  "textAlign": "center",
                },
                Object {},
              ],
            ]
          }
        >
          CREATE YOUR OWN STEP
        </Text>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
